packages/cli/src/lib/deploy/PolywrapDeploy.ts

```diff
@@ -0,0 +1,96 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+
+import { Deployer } from "./deployer";
+import { loadDeployManifest, loadDeployManifestExt } from "../project";
+import { CacheDirectory } from "../CacheDirectory";
+import { Logger } from "../logging";
+
+import { DeployManifest } from "@polywrap/polywrap-manifest-types-js";
+import { Schema as JsonSchema } from "jsonschema";
+import path from "path";
+import nodePath from "path";
+
+interface PolywrapDeployConfig {
+  cache: CacheDirectory;
+  logger: Logger;
+  defaultModulesCached: boolean;
+}
+
+export class PolywrapDeploy {
+  public static cacheLayout = {
+    root: "deploy/",
+    deployModulesDir: "modules/",
+  };
+  public manifest: DeployManifest;
+  private _config: PolywrapDeployConfig;
+
+  private constructor(
+    manifest: DeployManifest,
+    cache: CacheDirectory,
+    logger: Logger
+  ) {
+    this._config = {
+      logger,
+      cache,
+      defaultModulesCached: false,
+    };
+    this.manifest = manifest;
+  }
+
+  public static async create(
+    manifest: string,
+    logger: Logger
+  ): Promise<PolywrapDeploy> {
+    const deployManifest = await loadDeployManifest(manifest, logger);
+    const cache = new CacheDirectory({
+      rootDir: nodePath.dirname(manifest),
+      subDir: PolywrapDeploy.cacheLayout.root,
+    });
+    return new PolywrapDeploy(deployManifest, cache, logger);
+  }
+
+  public async getDeployModule(
```

Comment:
Related to my naming comment. We're literally returning a `Deployer` (a.k.a. DeployModule) here.

While thinking about this, we can also just open an issue for this and do it separately as to not overly increase the scope of this PR.

Reply:
hey i opened the changes in other pr on top of this https://github.com/polywrap/toolchain/pull/1483